lib/solidus_importer/processors/variant.rb

```diff
@@ -3,32 +3,47 @@
 module SolidusImporter
   module Processors
     class Variant < Base
+      attr_accessor :product
+
       def call(context)
         @data = context.fetch(:data)
-        return unless variant?
+        self.product = context.fetch(:product) || raise(ArgumentError, 'missing :product in context')
 
-        product = context.fetch(:product)
-        variant = process_variant(product)
-        context.merge!(variant: variant)
+        context.merge!(variant: process_variant)
       end
 
       private
 
-      def prepare_variant(product)
-        Spree::Variant.find_or_initialize_by(sku: @data['Variant SKU']) do |variant|
+      def prepare_variant
+        @prepare_variant ||= Spree::Variant.find_or_initialize_by(sku: sku) do |variant|
```

Comment:
Looks like Shopify bases the variant identity on the product + the name of the first option-value, maybe we should do the same, e.g.:
- update the master variant if the incoming data is for a master-variant
- use "Option1 Value" to identify the variant and maybe also to generate the sku when missing (E.g.: `"#{product.slug}-#{@data["Option1 Value"].parameterize}"`)